File: exam/src/objects/Contract.ts
import { Document } from "./Document";

export class Contract extends Document {
    private author: string = "";
    private version: string = "";

    constructor (filename: string, metadata: Record <string, string>) {
        super(filename, metadata);
        this.author = metadata["author"];
        this.version = metadata["version"];
    }

    public getAuthor(): string {
        return this.author;
    }

    public getVersion(): string {
        return this.version;
    }

    public setAuthor(author: string): void {
        this.author = author;
    }

    public setVersion(version: string): void {
        this.version = version;
    }
}

File: exam/src/documents/Contract.java
package documents;

import java.util.HashMap;

public class Contract extends Document {

    public Contract(String filename, HashMap<String, String> metadata) {
        super(filename, metadata);
    }

    @Override
    public boolean acceptedSize() {
        return Float.parseFloat(this.metadata.get("size")) < 3.00;
    }

    @Override
    public boolean acceptedExtensions() {
        return this.filename.toLowerCase().endsWith(".pdf");
    }

    @Override
    public boolean hasRequiredMetadata() {
        return this.metadata.containsKey("author") && this.metadata.containsKey("version");
    }

    @Override
    public boolean isValid() {
        return super.isValid() && acceptedSize() && acceptedExtensions() && hasRequiredMetadata();
    }
}
